viewing_party/party.py

```diff
@@ -1,23 +1,198 @@
+        return avg
+    return 0.0
+
+
+def get_most_watched_genre(user_data):
+    # first validate that user_data["watched"] is truthy, else return None
+    if not user_data["watched"]:
+        return None
+        
+    # store the nested list in its own variable for ease of access
+    watched_list = user_data["watched"]
+
+    # declare counter variables for the amount of movies in each genre
+    num_fantasy_movies = 0
+    num_action_movies = 0
+    num_intrigue_movies = 0
+
+    # iterate through the watched list movies to increment the corresponding genre count
+    for movie in watched_list:
+        if movie["genre"] == "Fantasy":
+            num_fantasy_movies += 1
+        elif movie["genre"] == "Action":
+            num_action_movies += 1
+        elif movie["genre"] == "Intrigue":
+            num_intrigue_movies += 1
+
+    most_watched = max(num_action_movies, num_fantasy_movies, num_intrigue_movies)
+
+    # find the corresponding genre to the most_watched amount
+    if num_action_movies == most_watched:
+        return "Action"
+    elif num_fantasy_movies == most_watched:
+        return "Fantasy"
+    elif num_intrigue_movies == most_watched:
+        return "Intrigue"
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+    friends_watched = []
+    user_unique = []
+    
+    # Create list with everyone else's movies
+    for friend in range(0, len(friends)):
+        friend_watched_movies = friends[friend]["watched"]
+        for movie in friend_watched_movies:
+            friends_watched.append(movie)
+
+    for film in user_watched:
+        if film not in friends_watched:
+            user_unique.append(film)
+
+    return user_unique
+
+def get_friends_unique_watched(user_data):
+    user_watched = user_data["watched"]
+    friends = user_data["friends"]
+    friends_watched = []
+    friends_unique = []
+    
+    for friend in range(0, len(friends)):
+        friend_watched_movies = friends[friend]["watched"]
+        print(f"friend_watched_movies: {friend_watched_movies}")        
+        for movie in friend_watched_movies:
+            if movie in friends_watched:
+                continue
+            friends_watched.append(movie)
+        print(f"friends_watched: {friends_watched}")
+    
+    for film in friends_watched:
+        if film not in user_watched:
+            friends_unique.append(film)
+    
+    print(f"friends_unique: {friends_unique}")
+    return friends_unique
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    available_recs = {}
+    user_watched_titles = []
+    for movie in user_data["watched"]:
+        if not (movie["title"] in user_watched_titles):

```

Comment:
Let's change this just a bit to make it more Pythonic
```suggestion
        if movie["title"] not in user_watched_titles:

```